java-client/src/main/java/org/opensearch/client/opensearch/_types/mapping/IcuCollationKeywordProperty.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch._types.mapping;
+
+import jakarta.json.stream.JsonGenerator;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+import org.opensearch.client.json.*;
+import org.opensearch.client.util.ObjectBuilder;
+
+// typedef: _types.mapping.ICUCollationKeywordProperty
+
+@JsonpDeserializable
+public class IcuCollationKeywordProperty extends DocValuesPropertyBase implements PropertyVariant {
```

Comment:
We probably should also support collation options [1] for these fields, right? 

[1] https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu-collation-keyword-field.html#_collation_options

Reply:
@reta good callout. I have added those now.